NitroxModel/Packets/Packet.cs

```diff
@@ -60,40 +65,64 @@ public enum UdpChannelId
 
         public byte[] Serialize()
         {
-            byte[] packetData;
-
-            using (MemoryStream ms = new MemoryStream())
-            using (LZ4Stream lz4Stream = new LZ4Stream(ms, LZ4StreamMode.Compress))
-            {
-                serializer.Serialize(lz4Stream, this);
-                packetData = ms.ToArray();
-            }
-
-            return packetData;
+            using MemoryStream ms = new MemoryStream();
+            using LZ4Stream lz4Stream = new LZ4Stream(ms, LZ4StreamMode.Compress);
+            serializer.Serialize(lz4Stream, this);
+            return ms.ToArray();
         }
 
         public static Packet Deserialize(byte[] data)
         {
-            using (Stream stream = new MemoryStream(data))
-            using (LZ4Stream lz4Stream = new LZ4Stream(stream, LZ4StreamMode.Decompress))
-            {
-                return (Packet)serializer.Deserialize(lz4Stream);
-            }
+            using Stream stream = new MemoryStream(data);
+            using LZ4Stream lz4Stream = new LZ4Stream(stream, LZ4StreamMode.Decompress);
+            return (Packet)serializer.Deserialize(lz4Stream);
         }
 
         public static bool IsTypeSerializable(Type type)
         {
             // We have our own surrogates to (de)serialize types that are not marked [Serializable]
             // This code is very similar to how serializability is checked in:
             // System.Runtime.Serialization.Formatters.Binary.BinaryCommon.CheckSerializable
-
-            ISurrogateSelector selector;
-            return (serializer.SurrogateSelector.GetSurrogate(type, Packet.serializer.Context, out selector) != null);
+            return serializer.SurrogateSelector.GetSurrogate(type, Packet.serializer.Context, out ISurrogateSelector _) != null;
         }
 
         public WrapperPacket ToWrapperPacket()
         {
             return new WrapperPacket(Serialize());
         }
+
+        public override string ToString()
+        {
+            Type packetType = GetType();
+
+            if (!cachedPropertiesByType.TryGetValue(packetType, out PropertyInfo[] properties))
+            {
+                properties = packetType.GetProperties(BindingFlags.Public | BindingFlags.Instance);
+                cachedPropertiesByType.Add(packetType, properties);
+            }
+
+            StringBuilder builder = new($"[{packetType.Name}: ");
+            foreach (PropertyInfo property in properties)
+            {
+                if (property.Name is nameof(DeliveryMethod) or nameof(UdpChannel))
+                {
+                    continue;
+                }
```

Comment:
These properties should be excluded in the cache to reduce impact when ToString() is called a lot.